### DISCUSSION TITLE
New Google Analytics?

### DISCUSSION THREAD

**Discussion by hardreddata** [TerriaJS/terriajs#5842] [Q&A]
Does anyone know about https://blog.google/products/marketingplatform/analytics/new_google_analytics/ ?

The email we got talks about updating `analytics.js` to `gtag.js`.

It sounds like the existing service will be retired. I don't have a feel for timeframes.

Thanks.

**Comment by tephenavies** [comment]
Thanks for putting that on our radar. If the old service properties are getting retired we'll have to update terriajs to use the new one. I haven't seen any timeframes either.

**Reply by zoran995** [reply]
Read about this about a month ago, there is no timeframe but probably we can expect it to stop working properly in 2022 when google disables all third-party cookies in chrome, and google analytics in the base is based on the cookie. I haven't look into possibility of data migration from old analytics
p.s. I would also consider adding a cookie consent while upgrading to GTM, currently terria uses
- analytics cookies, google analytic, youtube (unless user accept cookies we can't show videos),
- personalization cookies (storing values in local storage)

**Reply by imakihi** [reply]
GA on our terria instance will stop working after July 1st, 2023 (https://support.google.com/analytics/answer/11583528)? Do I need to change any settings to use GA4 instead of UA. I don't know much about this change but we are collecting stats using GA and want to use it after this change.
Can you tell me what I should do to adapt this change??
Thanks.

**Reply by tephenavies** [reply]
To switch to GA4 there will need to be changes to how the GA script tag is injected, how GA is initialised and it's possible we may also need to change how some events are fired. Following Google's migration guide to change the [`GoogleAnalytics.js` file](https://github.com/TerriaJS/terriajs/blob/main/lib/Core/GoogleAnalytics.js) should get you most of the way.

**Reply by imakihi** [reply]
Thank you for your comments. We started thinking about how to deal with this changes and we may be able to figure this out. If not, I may ask additional questions. Thanks again.

**Reply by zoran995** [reply]
@imakihi I created a base implementation back in October last year, but it got pushed to the side and honestly, I forgot about it. Tested it only locally through gtag debugger and here is a SS of one log.
![image](https://user-images.githubusercontent.com/19208948/172018113-2723f93d-4d38-4c6b-800e-c868fafabe91.png)

I just rebased it with the latest main version and you can find it here https://github.com/TerriaJS/terriajs/compare/main...zoran995:gtag

Idea was to implement it with a simple consent manager that will show first before setting up analytics and be globally available since youtube videos should not be visible to the user unless analytics cookies are accepted. That is probably more complex than you need right now, but EU users need it or have to use a third-party service for it. We might need to consider that the consent manager can be disabled and replaced with a third party service.

**Reply by imakihi** [reply]
@zoran995 Thank you for this info. I will take a look at this with our engneer (I am not an engineer).
If we need additional help, we may post questions again but thank you so much for helping.
Hiroo
